src/libstd/sys/unix/os.rs

```diff
@@ -328,6 +328,22 @@ pub fn current_exe() -> io::Result<PathBuf> {
 #[cfg(any(target_os = "linux", target_os = "android", target_os = "emscripten"))]
 pub fn current_exe() -> io::Result<PathBuf> {
     match crate::fs::read_link("/proc/self/exe") {
+        Ok(path) => {
+            let path_str = path.to_str().ok_or(io::Error::new(
+                io::ErrorKind::InvalidData,
+                "path contains invalid UTF-8 data",
```

Comment:
This seems likely to be incorrect... we shouldn't need to require a path to be UTF-8; that's the entire reason we have `Path(Buf)` and `OsStr(ing)`.